src/routes/console/project-[project]/storage/table.svelte

```diff
@@ -0,0 +1,149 @@
+        TableHeader,
+        TableRowLink,
+        TableScroll,
+        TableCellCheck
+    } from '$lib/elements/table';
+    import { toLocaleDateTime } from '$lib/helpers/date';
+    import { addNotification } from '$lib/stores/notifications';
+    import { sdk } from '$lib/stores/sdk';
+    import type { PageData } from './$types';
+    import { columns } from './store';
+
+    export let data: PageData;
+    const projectId = $page.params.project;
+
+    let selected: string[] = [];
+    let showDelete = false;
+    let deleting = false;
+
+    async function handleDelete() {
+        showDelete = false;
+
+        const promises = selected.map((databaseId) => sdk.forProject.databases.delete(databaseId));
+        try {
+            await Promise.all(promises);
+            trackEvent(Submit.DatabaseDelete);
+            addNotification({
+                type: 'success',
+                message: `${selected.length} database${selected.length > 1 ? 's' : ''} deleted`
+            });
+            invalidate(Dependencies.DATABASES);
+        } catch (error) {
+            addNotification({
+                type: 'error',
+                message: error.message
+            });
+            trackError(error, Submit.DatabaseDelete);
+        } finally {
+            selected = [];
+            showDelete = false;
+        }
+    }
+</script>
+
+<TableScroll>
+    <TableHeader>
+        <TableCellHeadCheck bind:selected pageItemsIds={data.buckets.buckets.map((c) => c.$id)} />
+        {#each $columns as column}
+            {#if column.show}
+                <TableCellHead width={column.width}>{column.title}</TableCellHead>
+            {/if}
+        {/each}
+    </TableHeader>
+    <TableBody>
+        {#each data.buckets.buckets as bucket}
+            <TableRowLink
+                href={`${base}/console/project-${projectId}/storage/bucket-${bucket.$id}`}>
+                <TableCellCheck bind:selectedIds={selected} id={bucket.$id} />
+                {#each $columns as column}
+                    {#if column.show}
+                        {#if column.id === '$id'}
+                            {#key $columns}
+                                <TableCell width={column.width} title={column.title}>
+                                    <Id value={bucket.$id}>
+                                        {bucket.$id}
+                                    </Id>
+                                </TableCell>
+                            {/key}
+                        {:else if column.id === 'name'}
+                            <TableCellText width={column.width} title={column.title}>
+                                {bucket.name}
+                            </TableCellText>
+                        {:else}
+                            <TableCellText width={column.width} title={column.title}>
+                                {toLocaleDateTime(bucket[column.id])}
+                            </TableCellText>
+                        {/if}
```

Comment:
this part is ugly and scales really bad if we would apply this to all screens. We should think about improving that part